# Proof Of Concept ML modeling

In [2]:
import pandas as pd
import numpy as np


df = pd.read_csv('../data/flats.csv')
df.head()

month        town flat_type block        street_name storey_range  \
0  2017-01  ANG MO KIO    2 ROOM   406  ANG MO KIO AVE 10     10 TO 12   
1  2017-01  ANG MO KIO    3 ROOM   108   ANG MO KIO AVE 4     01 TO 03   
2  2017-01  ANG MO KIO    3 ROOM   602   ANG MO KIO AVE 5     01 TO 03   
3  2017-01  ANG MO KIO    3 ROOM   465  ANG MO KIO AVE 10     04 TO 06   
4  2017-01  ANG MO KIO    3 ROOM   601   ANG MO KIO AVE 5     01 TO 03   

   floor_area_sqm      flat_model  lease_commence_date     remaining_lease  \
0            44.0        Improved                 1979  61 years 04 months   
1            67.0  New Generation                 1978  60 years 07 months   
2            67.0  New Generation                 1980  62 years 05 months   
3            68.0  New Generation                 1980   62 years 01 month   
4            67.0  New Generation                 1980  62 years 05 months   

   resale_price  
0      232000.0  
1      250000.0  
2      262000.0  
3      265000.0  
4      265000.0

In [3]:
df.describe()

floor_area_sqm  lease_commence_date  resale_price
count   181262.000000        181262.000000  1.812620e+05
mean        97.152903          1996.013842  4.978643e+05
std         24.025926            14.041087  1.726320e+05
min         31.000000          1966.000000  1.400000e+05
25%         82.000000          1985.000000  3.700000e+05
50%         93.000000          1996.000000  4.680000e+05
75%        112.000000          2010.000000  5.920000e+05
max        249.000000          2020.000000  1.588000e+06

In [4]:
batches = np.array_split(df, 5)

df = pd.DataFrame(batches[0])

/Users/maxmartyshov/Desktop/IU/year 2/sem3/MLOps-project/venv/lib/python3.12/site-packages/numpy/core/fromnumeric.py:59: FutureWarning: 'DataFrame.swapaxes' is deprecated and will be removed in a future version. Please use 'DataFrame.transpose' instead.
  return bound(*args, **kwds)


In [5]:
X = df[['floor_area_sqm', 'lease_commence_date']]
y = df['resale_price']

X.shape

(36253, 2)

In [6]:
from sklearn.linear_model import LinearRegression # base
from sklearn.linear_model import SGDRegressor # more complex with regularization
from sklearn.linear_model import RANSACRegressor # for outliers



lr = LinearRegression()
sdg = SGDRegressor()
ransac = RANSACRegressor()

In [7]:
lr.fit(X, y)
ransac.fit(X, y)

RANSACRegressor()

In [9]:
print('Linear regression mean accuracy:', lr.score(X, y))
print('RANSAC regression mean accuracy:', ransac.score(X, y))

Linear regression mean accuracy: 0.43034773066968435
RANSAC regression mean accuracy: 0.13437529431739803


### Conclusion:

In this proof-of-concept study, we trained a basic machine learning model using only two numerical features and without implementing any preprocessing or parameter tuning. Our initial results yielded accuracies of 0.43 and 0.13, indicating room for substantial improvement.

These modest accuracy levels are expected given the minimalistic approach taken. They serve as a baseline, highlighting the model's potential and confirming that machine learning can be applied to our project. These initial results are promising because they establish a starting point for further exploration.

To enhance model performance, future steps will include adding more features, applying data preprocessing techniques, and fine-tuning parameters. These enhancements will likely result in significant improvements in accuracy and overall model effectiveness.

Based on these preliminary findings, it is clear that machine learning holds promise for this project. We should therefore move forward with more advanced development and refinement to fully realize its potential.
